pkg/azclient/policy/ratelimit/ratelimit.go

```diff
@@ -73,3 +73,29 @@ func (f RateLimitPolicy) Do(req *policy.Request) (*http.Response, error) {
 	}
 	return req.Next()
 }
+
+// CloudProviderRateLimitConfig indicates the rate limit config for each clients.
+type CloudProviderRateLimitConfig struct {
+	// The default rate limit config options.
+	Config
+
+	// Rate limit config for each clients. Values would override default settings above.
+	Entries map[string]*Config `json:",inline" yaml:",inline"`
+}
+
+func NewCloudProviderRateLimitConfig() *CloudProviderRateLimitConfig {
+	return &CloudProviderRateLimitConfig{
+		Config: Config{
+			CloudProviderRateLimit: false,
+		},
+		Entries: make(map[string]*Config),
+	}
+}
+
+// GetRateLimitConfig returns the rate limit config for the given client. if the client is not found, the default is returned.
+func (config *CloudProviderRateLimitConfig) GetRateLimitConfig(clientName string) *Config {
```

Comment:
Do we have a Setter?

Reply:
the config is parsed from config json. We don't change it